static/app/utils/featureObserver.ts

```diff
@@ -1,71 +1,109 @@
   }) {
-    const FLAGS = this.FEATURE_FLAGS;
+    // Check if the flag is already in the buffer
+    const index = flagBuffer.values.findIndex(f => f.flag === flagName);
+
+    // The flag is already in the buffer
+    if (index !== -1) {
+      flagBuffer.values.splice(index, 1);
+    }
+
+    // If at capacity, we need to remove the earliest flag
+    // This will only happen if not a duplicate flag
+    if (flagBuffer.values.length === bufferSize) {
+      flagBuffer.values.shift();
+    }
+
+    // Store the flag and its result in the buffer
+    flagBuffer.values.push({
+      flag: flagName,
+      result: flagResult,
+    });
+  }
 
+  public observeOrganizationFlags({organization}: {organization: Organization}) {
+    const flagBuffer = this.FEATURE_FLAGS;
+    const bufferSize = this._bufferSize;
+    const updateFlagBuffer = this.updateFlagBuffer;
     // Track names of features that are passed into the .includes() function.
     const handler = {
-      apply: function (target, orgFeatures, flagName) {
+      apply: (target: any, orgFeatures: string[], flagName: string[]) => {
         // Evaluate the result of .includes()
         const flagResult = target.apply(orgFeatures, flagName);
 
         // Append `feature.organizations:` in front to match the Sentry options automator format
         const name = 'feature.organizations:' + flagName[0];
 
-        // Check if the flag is already in the buffer
-        const index = FLAGS.values.findIndex(f => f.flag === name);
+        updateFlagBuffer({flagName: name, flagResult, flagBuffer, bufferSize});
+
+        return flagResult;
+      },
+    };
+    const proxy = new Proxy(organization.features.includes, handler);
+    organization.features.includes = proxy;
+  }
 
-        // The flag is already in the buffer
-        if (index !== -1) {
-          FLAGS.values.splice(index, 1);
-        }
+  public observeProjectFlags({project}: {project: Project}) {
+    const flagBuffer = this.FEATURE_FLAGS;
+    const bufferSize = this._bufferSize;
+    const updateFlagBuffer = this.updateFlagBuffer;
+    // Track names of features that are passed into the .includes() function.
+    const handler = {
+      apply: function (target: any, projFeatures: string[], flagName: string[]) {
+        // Evaluate the result of .includes()
+        const flagResult = target.apply(projFeatures, flagName);
 
-        // If at capacity, we need to remove the earliest flag
-        // This will only happen if not a duplicate flag
-        if (FLAGS.values.length === bufferSize) {
-          FLAGS.values.shift();
-        }
+        // Append `feature.projects:` in front to match the Sentry options automator format
+        const name = 'feature.projects:' + flagName[0];
 
-        // Store the flag and its result in the buffer
-        FLAGS.values.push({
-          flag: name,
-          result: flagResult,
-        });
+        updateFlagBuffer({flagName: name, flagResult, flagBuffer, bufferSize});
```

Comment:
`this.updateFlagBuffer(...)` would work if `apply: function (....) {` were changed to arrow function syntax.

The [arrow function syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#cannot_be_used_as_methods) are bound statically   before runtime, and the `this` value is the same as the scope where the method is defined, so in this case the `this` inside `apply: () => { ... }` would be the same as the scope inside `public observeProjectFlags({project}: {project: Project}) { ... }` with `this` pointing to the instance of the class. woah!